### PR TITLE
Kitbot example

### PR DESCRIPTION
**Pull request by Lewis-Seiden**

### CODE REVIEW

File: KitbotExampleWalkthrough.md [hunk 1]
```diff
@@ -0,0 +1,245 @@
...
+For the `DoubleSuppliers` to the arcade drive command we can add two lambdas that call `getLeftY()` and `getRightX()` on the controller, which looks at the current joysick values.
+This means that each loop this command is run, it will check the current joystick values and turn them into voltages for the wheels.
+
+This line is a little long now, so break it up into a few lines to keep it readable.
+
+```Java
+drivetrainSubsystem.setDefaultCommand(
+    drivetrainSubsystem.setVoltagesArcadeCommand(
+        () -> controller.getLeftY(), 
+        () -> controller.getRightX()));
```
Comment (kevinclark) [inline KitbotExampleWalkthrough.md:201]: You can use a [method reference](https://docs.oracle.com/javase/tutorial/java/javaOO/methodreferences.html): `controller::getLeftY` and `controller::getRightX` and skip the extra lambda syntax.

`drivetrainSubsystem.setVoltagesArcadeCommand(controller::getLeftY, controller::getRightX)`

File: KitbotExampleWalkthrough.md [hunk 2]
```diff
@@ -0,0 +1,245 @@
...
+```
+
+Using the joystick values directly is perfectly fine, but sometimes our driver wants a different feel to the controls.
+One common way is to square the inputs to the drivetrain, which reduces the sensitivity of the drivebase at low speeds.
+Another is to add a "deadband" where we wont output any voltage if the input is less than some threshold.
+This stops the robot from moving if the controllers don't perfectly read 0 when they aren't pressed.
+Lets add a function in RobotContainer to modify the joystick inputs.
```
Comment (kevinclark) [inline KitbotExampleWalkthrough.md:208]: When I explain this I find that drawing the graph often helps a lot. You might want a picture here.

**Review by kevinclark — CHANGES_REQUESTED** [review]
This is really great. I worry a bit about the learning curve here. There are things that probably need an in person explanation or simple lecture. There's also stuff in the example code that isn't touched on (how the Commands get scheduled, what's with this Robot class, etc) but I think it's totally reasonable to leave that for later. This is already great content that's going to work for a lot of people.

Has the code actually been run on a robot (or at least a sim) to make sure that it compiles and functions properly?

File: KitbotExampleWalkthrough.md (hunk 1 above)
Comment (Lewis-Seiden) [reply KitbotExampleWalkthrough.md:201]: I avoided that since its more new syntax and because I'm going to wrap these in a method call later.

**Review by kevinclark — APPROVED** [review]

**Pull request merged**